src/Provisioning.ts

```diff
@@ -282,6 +282,57 @@ export class Provisioner {
         });
     }
 
+    @command("user_id", "channel_id", "team_id", "bot_token")
+    private async channelinfo(_, res, userId, channelId, teamId, botToken) {
+        if (typeof userId !== 'string' ||
+            typeof channelId !== 'string' ||
+            typeof teamId !== 'string' ||
+            typeof botToken !== 'string') {
+            return res.status(HTTP_CODES.CLIENT_ERROR).json({
+                message: 'user_id, channel_id, team_id and bot_token must be strings',
+            });
+        }
+
+        const opts = {
+            slack_bot_token: botToken,
+            slack_channel_id: channelId,
+            team_id: teamId,
+        };
+
+        log.info(`${userId} requested the room info of ${opts.slack_channel_id}`);
+
+        // Check if the user is in the team.
+        if (!(await this.main.matrixUserInSlackTeam(opts.team_id, userId))) {
+            return Promise.reject({
+                code: HTTP_CODES.FORBIDDEN,
+                text: `${userId} is not in this team.`,
+            });
+        }
+
+        let channelInfo;
+        try {
+            channelInfo = await this.main.getChannelInfo(opts);
```

Comment:
You could simplyify this code a lot by having the res.json part inside the try catch statement, 
```suggestion
            const channelInfo = await this.main.getChannelInfo(opts);
            if (!channelInfo) {
                return res.status(HTTP_CODES.NOT_FOUND).json({
                    message: 'Slack channel not found',
                });
            }
            res.json({
                name: channelInfo.channel.name,
                memberCount: channelInfo.channel.num_members,
            });
```

Reply:
Applied suggestion.